### DISCUSSION TITLE
Regex in safelist

### DISCUSSION THREAD

**Discussion by ljcremer** [windicss/windicss#32] [Q&A] [answered]
Is it possible to use a regex expression in the safelist?

**Comment by antfu** [comment, marked as answer]
I don't think so actually. Unlike Tailwind, Windi builds CSS on-demand, which means there are infinitely many possible classes for a single regex like: `p-*` for `p-1` `p-1.1` `p-1.2`...

**Reply by ljcremer** [reply]
Thanks. 

What would you recommend would be the best way to get a number of custom and dynamic utility classes into the safelist?

**Reply by antfu** [reply]
Maybe something like this?

```ts
{
  safelist: [
    ...new Array(50).fill(0).map((, i) => `p-${i+1}`), // for p-1 p-2 ... p-50
    ...new Array(100).fill(0).filter((, i) => (i + 1) % 2 === 0).map((, i) => `m-${i+1}`), // for m-2 m-4 m-6 ... m-100
  ]   
}
```

**Reply by ljcremer** [reply]
Great thanks. Let me play around with that a bit and see what I can get working
